userspace/libscap/ringbuffer/ringbuffer.c

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <math.h>
+#include <scap.h>
+#include "../../../driver/ppm_ringbuffer.h"
+
+/* Dimension of a single per-CPU buffer in bytes. */
+unsigned long global_buffer_bytes_dim;
```

Comment:
Does this really have to be a global? It feels like it belongs in the scap_device struct (or scap_device_set but that would be less convenient).

We already have scap_device.m_buffer_size, how about we use that? It means we can't have a "set the size globally" function like the one below but tbh I don't see the downside in that :P 

The technical limitation is that a particular ring buffer can't have its size changed online, everything else is possible (even if pointless) so how about we express that fact in the code?

Also, I realized that the current udig code is probably broken (by my patches) precisely because it doesn't seem to use m_buffer_size anywhere, contrary to what the comment says. So using that field instead of the global would fix two issues at the same time.

Reply:
yeah this is a smart idea as always, I have thought about that, the only problem is that actually this variable contains the memory mapped size and is used to `unmap` the memory during the closing phase see an example here (https://github.com/falcosecurity/libs/blob/a57b941be6702eb8459c83cfcfec0397ea8fea81/userspace/libscap/engine/kmod/scap_kmod.c#L274). Let's say, as @incertum highlighted this is a first step to support the variable size dim, but yes we can also improve things in following refactors. The problem is always the same we have used a lot of stuff without a precise scope and now we have 10 methods to do the same things :/ But completely agree we need to improve the situation here, probably in this PR since we need this for the release

Reply:
> this variable contains the memory mapped size and is used to unmap the memory

we can always `munmap(addr, size * 2)` :)

Reply:
BTW I think that the right place is `scap_device_set` since this dimension is shared between all the devices